embulk-output-redshift/src/main/java/org/embulk/output/redshift/RedshiftCopyBatchInsert.java

```diff
@@ -86,28 +93,37 @@ public void flush() throws IOException, SQLException
     {
         File file = closeCurrentFile();  // flush buffered data in writer
 
-        // TODO multi-threading
-        new UploadAndCopyTask(file, batchRows, s3KeyPrefix + UUID.randomUUID().toString()).call();
-        new DeleteFileFinalizer(file).close();
+        UploadAndCopyTask task = new UploadAndCopyTask(file, batchRows, s3KeyPrefix + UUID.randomUUID().toString());
+        executorService.submit(task);
```

Comment:
I think error handling is necessary to stop a bulk import job appropriately when upload or COPY failed.
`ExecutorService#submit()` returns `Future<Void>` and `Future#get()` throws ExecutionException if the callable throws an exception.